pontoon/teams/views.py

```diff
@@ -379,6 +384,90 @@ def ajax_translation_memory_delete(request, locale):
     return HttpResponse("ok")
 
 
+@require_AJAX
+@require_POST
+@permission_required_or_403("base.can_translate_locale", (Locale, "code", "locale"))
+@transaction.atomic
+def ajax_translation_memory_upload(request, locale):
+    """Upload Translation Memory entries from a .TMX file."""
+    try:
+        file = request.FILES["tmx_file"]
+    except MultiValueDictKeyError:
+        return JsonResponse(
+            {"status": False, "message": "No file uploaded."},
+            status=400,
+        )
+
+    if not file.name.endswith(".tmx"):
+        return JsonResponse(
+            {
+                "status": False,
+                "message": "Invalid file format. Only .TMX files are supported.",
+            },
+            status=400,
+        )
+
+    locale = get_object_or_404(Locale, code=locale)
+    code = locale.code
+
+    # Parse the TMX file
+    try:
+        tree = ET.parse(file)
+        root = tree.getroot()
+    except ET.ParseError as e:
+        return JsonResponse(
+            {"status": False, "message": f"Invalid XML file: {e}"}, status=400
+        )
+
+    # Extract TM entries
+    file_entries = []
+    ns = {"xml": "http://www.w3.org/XML/1998/namespace"}
+    for tu in root.findall(".//tu"):
+        try:
+            source = tu.find("./tuv[@xml:lang='en-US']/seg", namespaces=ns).text.strip()
+            target = tu.find(
+                f"./tuv[@xml:lang='{code}']/seg", namespaces=ns
+            ).text.strip()
+
+            if source and target:
+                file_entries.append({"source": source, "target": target})
+        except Exception as e:
+            log.info(f"Error processing <tu>: {e}")
+
+    if not file_entries:
+        return JsonResponse(
+            {"status": False, "message": "No valid translation entries found."},
+            status=400,
+        )
+
+    # Create TranslationMemoryEntry objects
+    tm_entries = [
+        TranslationMemoryEntry(
+            source=entry["source"],
+            target=entry["target"],
+            locale=locale,
+        )
+        for entry in file_entries
+    ]
+
+    created_entries = TranslationMemoryEntry.objects.bulk_create(
+        tm_entries, batch_size=1000
+    )
```

Comment:
Should we ignore duplicates? There aren't many downsides of not doing so, safe for making the TranslationMemoryEntry larger.

Reply:
> There aren't many downsides of not doing so, safe for making the TranslationMemoryEntry larger.

You lost me at the double negative. The only downside is making TranslationMemoryEntry larger? But why would we want that, if we can avoid it?

Reply:
Not ignoring dupes makes the code simpler and faster and thus the max. file upload size larger. 

It's also less confusing than uploading a TMX with 1000 entries and getting a "531 entries uploaded" success message. But we can include the number of skipped entries in the message as well.

I'll take a stab at that.

Reply:
Updated.